src/components/distributions/editor/TextForm/TextInput.js

```diff
@@ -73,7 +77,7 @@ export default class TextInput extends Component{
     value: PropTypes.string,
   }
 
-  componentWillUnmount() { this._handleBlur() }
+  componentWillUnmount() { this._handleBlur(this.props.value) }
```

Comment:
Yes, it is needed. This is used when the whole form unblurs during dismount (like you click outside the metric card).